tests/Helpers/LogAnalyticsHelper.cs

```diff
@@ -0,0 +1,206 @@
+using Azure.Monitor.Ingestion;
+using AzureMcp.Services.Azure.Authentication;
+using AzureMcp.Services.Interfaces;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace AzureMcp.Tests.Helpers;
+
+/// <summary>
+/// Helper class for sending logs to Azure Log Analytics using the Azure Monitor Ingestion SDK.
+/// </summary>
+/// <remarks>
+/// Initializes a new instance of the LogAnalyticsHelper class.
+/// </remarks>
+public class LogAnalyticsHelper(
+    string workspaceName,
+    string subscription,
+    IMonitorService monitorService,
+    string? tenantId = null,
+    string logType = "TestLogs_CL",
+    ILogger? logger = null)
+{
+    private readonly string _workspaceName = workspaceName;
+    private readonly string _subscription = subscription;
+    private readonly string _logType = logType;
+    private readonly string? _tenantId = tenantId;
+    private readonly TokenCredential _credential = new CustomChainedCredential(tenantId);
+    private readonly IMonitorService _monitorService = monitorService ?? throw new ArgumentNullException(nameof(monitorService));
+    private readonly ILogger _logger = logger ?? NullLogger.Instance;
+    private readonly SemaphoreSlim _clientInitLock = new(1, 1);
+    private string? _workspaceId;
+    private LogsIngestionClient? _logsIngestionClient;
+
+    private async Task<string> GetWorkspaceIdAsync()
+    {
+        if (!string.IsNullOrEmpty(_workspaceId))
+        {
+            return _workspaceId;
+        }
+
+        // Get workspace info using the monitor service
+        var workspaces = await _monitorService.ListWorkspaces(_subscription, _tenantId);
+        var workspace = workspaces.FirstOrDefault(w => w.Name.Equals(_workspaceName, StringComparison.OrdinalIgnoreCase))
+            ?? throw new InvalidOperationException($"Could not find workspace {_workspaceName}");
+
+        _workspaceId = workspace.CustomerId;
+        return _workspaceId;
+    }
+
+    private async Task<LogsIngestionClient> GetLogsIngestionClientAsync(string customerId)
+    {
+        if (_logsIngestionClient != null)
+        {
+            return _logsIngestionClient;
+        }
+
+        if (string.IsNullOrEmpty(customerId))
+        {
+            throw new ArgumentNullException(nameof(customerId), "Customer ID cannot be null or empty");
+        }
+
+        await _clientInitLock.WaitAsync();
+        try
+        {
+            // Double-check after acquiring lock
+            if (_logsIngestionClient != null)
+            {
+                return _logsIngestionClient;
+            }
+
+            var endpoint = new Uri($"https://{customerId}.ods.opinsights.azure.com");
+            var options = new LogsIngestionClientOptions
+            {
+                Retry =
+                {
+                    MaxRetries = 3,
+                    Delay = TimeSpan.FromSeconds(2),
+                    MaxDelay = TimeSpan.FromSeconds(10)
+                }
+            };
```

Comment:
nit: Do we need a custom retry configuration? It would be good to add a comment why the default retry configuration won't work.

Reply:
That is what I think is best for the test.